crates/web5/src/apid/credentials/verifiable_credential_1_1.rs

```diff
@@ -98,26 +171,302 @@ impl VerifiableCredential {
+        }
+
+        if sub != vc.credential_subject.id {
+            return Err(CredentialError::ClaimMismatch("subject".to_string()));
+        }
+
+        // disregard nano-seconds which may be slightly different as a result of different time libraries
+        if nbf.duration_since(UNIX_EPOCH)?.as_secs()
+            != vc.issuance_date.duration_since(UNIX_EPOCH)?.as_secs()
+        {
+            return Err(CredentialError::ClaimMismatch("issuance_date".to_string()));
+        }
+
+        if let Some(exp) = exp {
+            if SystemTime::now() > exp {
+                return Err(CredentialError::CredentialExpired);
+            }
+        }
+
+        // TODO prioritize JWT claims -- hit up Neal
+
+        validate_vc_data_model(&vc)?;
+
+        Ok(vc)
+    }
+}
+
+fn validate_vc_data_model(vc: &VerifiableCredential) -> Result<()> {
+    // Required fields ["@context", "id", "type", "issuer", "issuanceDate", "credentialSubject"]
+    if vc.id.is_empty() {
+        return Err(CredentialError::VcDataModelValidationError(
+            "missing id".to_string(),
+        ));
+    }
+
+    if vc.context.is_empty() || vc.context[0] != BASE_CONTEXT {
+        return Err(CredentialError::VcDataModelValidationError(
+            "missing context".to_string(),
+        ));
+    }
+
+    if vc.r#type.is_empty() || vc.r#type[0] != BASE_TYPE {
+        return Err(CredentialError::VcDataModelValidationError(
+            "missing type".to_string(),
+        ));
+    }
+
+    if vc.issuer.to_string().is_empty() {
+        return Err(CredentialError::VcDataModelValidationError(
+            "missing issuer".to_string(),
+        ));
+    }
+
+    if vc.credential_subject.id.is_empty() {
+        return Err(CredentialError::VcDataModelValidationError(
+            "missing credential subject".to_string(),
+        ));
+    }
+
+    let now = SystemTime::now();
+    if vc.issuance_date > now {
+        return Err(CredentialError::VcDataModelValidationError(
+            "issuance date in future".to_string(),
+        ));
+    }
+
+    // Validate expiration date if it exists
+    if let Some(expiration_date) = &vc.expiration_date {
+        if expiration_date < &now {
+            return Err(CredentialError::VcDataModelValidationError(
+                "credential expired".to_string(),
+            ));
+        }
+    }
+
+    // TODO: Add validations to credential_status, credential_schema, and evidence once they are added to the VcDataModel
+    // https://github.com/TBD54566975/web5-rs/issues/112
+
+    Ok(())
+}
+
+#[derive(Clone)]
+pub struct JoseSigner {
+    pub kid: String,
+    pub signer: Arc<dyn Signer>,
+}
+
+impl JosekitJwsSigner for JoseSigner {
+    fn algorithm(&self) -> &dyn JosekitJwsAlgorithm {
+        &JosekitEddsaJwsAlgorithm::Eddsa
+    }
+
+    fn key_id(&self) -> Option<&str> {
+        Some(&self.kid)
+    }
+
+    fn signature_len(&self) -> usize {
+        64
```

Comment:
I think there was a JoseKit Enum for this

Reply:
there is `EdCurve::Ed25519` but it doesn't have the usize value associated to it